systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java

```diff
@@ -4,28 +4,254 @@
+import io.vertx.core.json.JsonObject;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+import static io.strimzi.api.kafka.model.KafkaResources.kafkaStatefulSetName;
 import static io.strimzi.systemtest.Constants.REGRESSION;
 import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
 
 @Tag(REGRESSION)
 public class CruiseControlST extends BaseST {
 
     private static final Logger LOGGER = LogManager.getLogger(CruiseControlST.class);
+    private static final String NAMESPACE = "cruise-control-test";
+
+    private static final String CRUISE_CONTROL_NAME = "Cruise Control";
+    private static final int CRUISE_CONTROL_DEFAULT_PORT = 9090;
+    private static final String CRUISE_CONTROL_ENDPOINT = "/kafkacruisecontrol/state";
+    private static final String CRUISE_CONTROL_POD_PREFIX = CLUSTER_NAME + "-cruise-control-";
+
+    private static final String CRUISE_CONTROL_CAPACITY_FILE_PATH = "/tmp/capacity.json";
+    private static final String CRUISE_CONTROL_CONFIGURATION_FILE_PATH = "/tmp/cruisecontrol.properties";
+
+    private static final String CRUISE_CONTROL_METRICS_TOPIC = "strimzi.cruisecontrol.metrics"; // partitions 1 , rf - 1
+    private static final String CRUISE_CONTROL_MODEL_TRAINING_SAMPLES_TOPIC = "strimzi.cruisecontrol.modeltrainingsamples"; // partitions 32 , rf - 2
+    private static final String CRUISE_CONTROL_PARTITION_METRICS_SAMPLES_TOPIC = "strimzi.cruisecontrol.partitionmetricsamples"; // partitions 32 , rf - 2
+
+    @Test
+    void testCruiseControlDeployment()  {
+        String ccStatusCommand = "curl -X GET localhost:" + CRUISE_CONTROL_DEFAULT_PORT + CRUISE_CONTROL_ENDPOINT;
+
+        String ccPodName = PodUtils.getFirstPodNameContaining("cruise-control");
+
+        String result = cmdKubeClient().execInPodContainer(ccPodName, "cruise-control",
+            "/bin/bash", "-c", ccStatusCommand).out();
+
+        LOGGER.info("Verifying that {} is running inside the Pod {} and REST API is available", CRUISE_CONTROL_NAME, ccPodName);
+
+        assertThat(result, not(containsString("404")));
+        assertThat(result, containsString("RUNNING"));
```

Comment:
As far as I remember the state endpoint return the status of various services running or not on CC. Maybe just checking that Running exists is not enough to test that CC is working properly.

Reply:
Some suggestion what is needed to verify?

Reply:
this is an example of response ...

```json
{
  "AnalyzerState": {
    "isProposalReady": false,
    "readyGoals": []
  },
  "MonitorState": {
    "trainingPct": 0,
    "trained": false,
    "numFlawedPartitions": 0,
    "state": "RUNNING",
    "numTotalPartitions": 65,
    "numMonitoredWindows": 0,
    "monitoringCoveragePct": 0,
    "reasonOfLatestPauseOrResume": "N/A",
    "numValidPartitions": 0
  },
  "ExecutorState": {
    "state": "NO_TASK_IN_PROGRESS"
  },
  "AnomalyDetectorState": {
    "recentBrokerFailures": [],
    "recentGoalViolations": [],
    "selfHealingDisabled": [
      "DISK_FAILURE",
      "BROKER_FAILURE",
      "METRIC_ANOMALY",
      "GOAL_VIOLATION",
      "TOPIC_ANOMALY"
    ],
    "balancednessScore": 100,
    "selfHealingEnabled": [],
    "recentDiskFailures": [],
    "metrics": {
      "meanTimeToStartFixMs": 0,
      "numSelfHealingStarted": 0,
      "ongoingAnomalyDurationMs": 0,
      "numSelfHealingFailedToStart": 0,
      "meanTimeBetweenAnomaliesMs": {
        "METRIC_ANOMALY": 0,
        "TOPIC_ANOMALY": 0,
        "DISK_FAILURE": 0,
        "GOAL_VIOLATION": 0,
        "BROKER_FAILURE": 0
      }
    },
    "recentMetricAnomalies": [],
    "recentTopicAnomalies": [],
    "selfHealingEnabledRatio": {
      "DISK_FAILURE": 0,
      "BROKER_FAILURE": 0,
      "METRIC_ANOMALY": 0,
      "GOAL_VIOLATION": 0,
      "TOPIC_ANOMALY": 0
    }
  },
  "version": 1
}
```

@tomncooper what do you think is the best field to check? Maybe I was wrong and checking the MonitorState is enough to check that state is RUNNING?

Reply:
So the load monitor state can be one of the following options:
```
public enum LoadMonitorTaskRunnerState {
  NOT_STARTED, RUNNING, PAUSED, SAMPLING, BOOTSTRAPPING, TRAINING, LOADING
}
```
Any of which (except maybe `NOT_STARTED`) show that CC is running. Are we just trying to verify that CC is up and running correctly? Basically if you get a JSON return from the State endpoint then CC is up. However if you want to check it is actually working you should probably check for the load monitor doing someting other than `NOT_STARTED` or `PAUSED`?

If you want to check functionality of an active rebalance you can check the `ExecutorState` which can have the following values:
``` 
  public enum State {
    NO_TASK_IN_PROGRESS,
    STARTING_EXECUTION,
    INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS,
    INTRA_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS,
    LEADER_MOVEMENT_TASK_IN_PROGRESS,
    STOPPING_EXECUTION
  }
```

Reply:
How can I invoke the situation where LoadMonitor will be in the state for instance in `PAUSED`. It would be great if I have some reproduction procedure. @tomncooper

Reply:
I don't know how you would get the load monitor to pause. It is not documented anywhere. But from reading the code it is a possible state.

Reply:
So instead of looking for just RUNNING. Look for any of RUNNING, SAMPLING, BOOTSTRAPPING, TRAINING, LOADING as these are all states that indicate that the load monitor is doing useful valid work.